### ISSUE TITLE
[BUG] No resource limits set for the Kubevirt's pod

### ISSUE THREAD

**Issue by guangbochen** [harvester/harvester#2729]
**Describe the bug**
 When one looks at the kubevirt pods in the `harvester-system` namespace, only the `virt-operator` pods seem to have limits set.

Additionally, the following might be related to this:
- The alerts KubeVirtComponentExceedsRequestedCPU and KubeVirtComponentExceedsRequestedMemory are not working properly

**To Reproduce**
1. Run `$kubectl get pod -n harvester-system`
2. Check kubevirt's related pod and only the virt-operator has resource limits

**Expected behavior**

Can you please confirm that :
- Putting requests on the Kubevirt pods is being worked on.
- Solving this would remove the alerts mentioned above

**Support bundle**

**Environment**
 - Harvester ISO version: v1.0.3
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): All

**Additional context**

**Comment by chrisho** [comment]
Look into the Kubevirt components CPU spike issue

**Comment by rthallisey** [comment]
Can you share some detail about the pressure you're applying to the KubeVirt control-plane, e.g. creating 100 vmis at once?

We have tooling for you to profile the control-plane.  Can you profile and share any interesting findings? https://github.com/kubevirt/kubevirt/tree/main/tools/cluster-profiler

**Comment by chrisho** [comment]
> Can you share some detail about the pressure you're applying to the KubeVirt control-plane, e.g. creating 100 vmis at once?

I have tried to restart 100 vmis at once, and I guess the same problem happens with creating 100vmis at the same time.

> We have tooling for you to profile the control-plane. Can you profile and share any interesting findings? https://github.com/kubevirt/kubevirt/tree/main/tools/cluster-profiler

thanks, will share with after the profile.

**Comment by chrisho** [comment]
Performance test report: https://github.com/harvester/harvester/pull/2774#issuecomment-1294426262

**Comment by davidspek** [comment]
For reference, here’s a good blog post about why cpu limits should generally not be set. https://home.robusta.dev/blog/stop-using-cpu-limits

**Comment by TachunLin** [comment]
Verified **fixed** on `master-6fd9a5bb-head` (2/10). Close this issue.

### Result
Verified by default Harvester already set resource limit to related Kubevirt's pods 

- `virt-api`, 
- `virt-controller` 
- `virt-handler`  

with the following resource limit value
  ```
  virt-api: {“cpu”:“400m”,“memory”:“1100Mi”}
  virt-controller: {“cpu”:“800m”,“memory”:“1300Mi”}
  virt-handler: {“cpu”:“700m”,“memory”:“1600Mi”}
  ```


```
$ kubectl get deploy virt-api -n harvester-system  -o yaml

resources:
          limits:
            cpu: 400m
            memory: 1100Mi
          requests:
            cpu: 5m
            memory: 150Mi
```

```
$ kubectl get deploy virt-controller -n harvester-system  -o yaml

resources:
          limits:
            cpu: 800m
            memory: 1300Mi
          requests:
            cpu: 10m
            memory: 150Mi

```

```
$ kubectl get ds virt-handler -n harvester-system  -o yaml

resources:
          limits:
            cpu: 700m
            memory: 1600Mi
          requests:
            cpu: 10m
            memory: 230M

```

### Test Information
* Test Environment: 1 node Harvester on local kvm machine 
* Harvester version: `master-6fd9a5bb-head` (2/10)

### Verify Steps
1. ssh to Harvester management node 
2. Check the kubevirt pods status
  ```
  $ kubectl get pod -n harvester-system | grep virt
  
  harv230213:~ # kubectl get pod -n harvester-system | grep virt
  virt-api-7cfd78b999-sk2s6                               1/1     Running   0          137m
  virt-controller-5d54b8b9bf-szh86                        1/1     Running   0          137m
  virt-controller-5d54b8b9bf-zz72s                        1/1     Running   0          137m
  virt-handler-6r8s8                                      1/1     Running   0          137m
  virt-operator-7bdb8d7bff-dtl7t                          1/1     Running   0          139m
  ```
3. Check the virt-api yaml content 
  ```
  kubectl get deploy virt-api -n harvester-system  -o yaml
  ```
4. Check the virt-controller yaml content  
  ```
  kubectl get deploy virt-controller -n harvester-system  -o yaml
  ```
5. Check the virt-handler yaml content 
  ```
  kubectl get ds virt-controller -n harvester-system  -o yaml
  ```
